shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java

```diff
@@ -44,23 +49,44 @@
     static final String ACTION_SUBMIT_FEEDBACK = "ActionSubmitFeedback";
     static final String ACTION_EDIT_IMAGE = "ActionEditImage";
     static final String EXTRA_USER_MESSAGE = "ExtraUserMessage";
+    static final String EXTRA_SUBCATEGORY = "ExtraSubcategory";
     // If you want to change the send icon, you must create a new menu
     static final @MenuRes int DEFAULT_MENU = R.menu.shaky_feedback_activity_actions;
 
     private static final String KEY_SCREENSHOT_URI = "ScreenshotUri";
     private static final String KEY_TITLE = "title";
     private static final String KEY_HINT = "hint";
     private static final String KEY_MENU = "menu";
+    private static final String KEY_SUBTYPE_LABELS = "subtypeLabels";
+    private static final String KEY_SUBTYPES = "subtypes";
 
+    /**
+     * Factory method for creating a {@link FormFragment}
+     *
+     * @param title The title of the form
+     * @param hint Hint to display in the feedback text form when it is empty
+     * @param screenshotUri {@link Uri} of the screenshot to attach if known
+     * @param menu Resource ID of the "Send" icon
+     * @param subtypeLabels Resource ID of the array of labels for subcategories to be shown in the
+     *                      spinner, if subcategories apply. The resulting array must be the same
+     *                      length and order as the "subtypes" param
+     * @param subtypes Subtypes of this feedback type, if subtypes apply. Must be the same length and
+     *                 order as the subtype labels array from the "subtypeLabels" param
+     * @return A new {@link FormFragment}
+     */
     public static FormFragment newInstance(@NonNull String title,
                                            @NonNull String hint,
                                            @Nullable Uri screenshotUri,
-                                           @Nullable @MenuRes int menu) {
+                                           @Nullable @MenuRes int menu,
+                                           @Nullable @ArrayRes int subtypeLabels,
+                                           @Nullable String[] subtypes) {
```

Comment:
I will not try to modify this public function, which introduces a breaking change on all downstream callers. Another newInstance method and delegate the call to this one should help

Reply:
Done